packages/ui-contexts/src/hooks/useVerifyPassword.ts

```diff
@@ -22,38 +20,44 @@ export const useVerifyPassword = (password?: string) => {
 		}
 
 		return true;
-	};
-
-	const passwordVerificationsTemplate: Record<string, (lengthCriteria?: number) => boolean> = {
-		'get-password-policy-minLength': (minLength?: number) => Boolean(minLength && password.length >= minLength),
-		'get-password-policy-maxLength': (maxLength?: number) => Boolean(maxLength && password.length <= maxLength),
-		'get-password-policy-forbidRepeatingCharactersCount': handleRepeatingChars,
-		'get-password-policy-mustContainAtLeastOneLowercase': () => /[a-z]/.test(password),
-		'get-password-policy-mustContainAtLeastOneUppercase': () => /[A-Z]/.test(password),
-		'get-password-policy-mustContainAtLeastOneNumber': () => /[0-9]/.test(password),
-		'get-password-policy-mustContainAtLeastOneSpecialCharacter': () => /[^A-Za-z0-9\s]/.test(password),
-	};
-
-	const passwordVerifications = {} as Record<string, { isValid: boolean; limit?: number }>;
-
-	data?.policy.forEach((currentPolicy) => {
-		if (!Array.isArray(currentPolicy)) return;
-
-		if (currentPolicy[0] === 'get-password-policy-forbidRepeatingCharacters') return;
-
-		if (currentPolicy[1]) {
-			passwordVerifications[currentPolicy[0]] = {
-				isValid: passwordVerificationsTemplate[currentPolicy[0]](Object.values(currentPolicy[1])[0]),
-				limit: Object.values(currentPolicy[1])[0],
-			};
-			return;
-		}
+	},
+	'get-password-policy-mustContainAtLeastOneLowercase': (password: string) => /[a-z]/.test(password),
+	'get-password-policy-mustContainAtLeastOneUppercase': (password: string) => /[A-Z]/.test(password),
+	'get-password-policy-mustContainAtLeastOneNumber': (password: string) => /[0-9]/.test(password),
+	'get-password-policy-mustContainAtLeastOneSpecialCharacter': (password: string) => /[^A-Za-z0-9\s]/.test(password),
+};
+
+type PasswordVerifications = Record<string, { isValid: boolean; limit?: number }>;
+type PasswordPolicies = [key: string, value?: Record<string, number>][];
+
+export const useVerifyPasswordByPolices = (policies: PasswordPolicies) => {
+	return useCallback(
+		(password: string) => {
+			return policies.reduce((passwordVerifications, [name, rules]) => {
+				if (name === 'get-password-policy-forbidRepeatingCharacters') return passwordVerifications;
+
+				if (rules) {
+					passwordVerifications[name] = {
+						isValid: password.length < 0 || passwordVerificationsTemplate[name](password, Object.values(rules)[0]),
```

Comment:
Is the verification `password.length < 0` correct? Strings cant have negative lengths and I believe the `watch('password')` function never returns a negative number so this check would be meaningless.